### PR TITLE
feat: upgrade to django 4.2

### PR DESCRIPTION
**Pull request by irtazaakram**
## Description

Describe what this pull request changes, and why. Include implications for people using this change.
Design decisions and their rationales should be documented in the repo (docstring / ADR), per
[OEP-19](https://open-edx-proposals.readthedocs.io/en/latest/oep-0019-bp-developer-documentation.html), and can be
linked here.

Useful information to include:
- Which edX user roles will this change impact? Common user roles are "Learner", "Course Author",
"Developer", and "Operator".
- Include screenshots for changes to the UI (ideally, both "before" and "after" screenshots, if applicable).
- Provide links to the description of corresponding configuration changes. Remember to correctly annotate these
changes.

## Supporting information

Link to other information about the change, such as Jira issues, GitHub issues, or Discourse discussions.
Be sure to check they are publicly readable, or if not, repeat the information here.

## Testing instructions

Please provide detailed step-by-step instructions for testing this change.

## Deadline

"None" if there's no rush, or provide a specific date or event (and reason) if there is one.

## Other information

Include anything else that will help reviewers and consumers understand the change.
- Does this change depend on other changes elsewhere?
- Any special concerns or limitations? For example: deprecations, migrations, security, or accessibility.
- If your [database migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) can't be rolled back easily.

### CODE REVIEW

File: problem_builder/tests/unit/test_answer_mixin.py [hunk 1]
```diff
@@ -25,7 +25,7 @@ def setUp(self):
 
     def make_answer_mixin(self, name=None, course_id=None, student_id=None):
         if name is None:
-            name = get_random_string()
+            name = get_random_string(12)
```
Comment (UsamaSadiq) [inline problem_builder/tests/unit/test_answer_mixin.py:28]: why is this change needed?

File: requirements/pip-tools.txt [hunk 2]
```diff
@@ -10,13 +10,14 @@ click==8.1.3
     # via pip-tools
 packaging==23.1
     # via build
-pip-tools==6.13.0
+pip-tools==6.14.0
```
Comment (UsamaSadiq) [inline requirements/pip-tools.txt]: let's bump the version again to `7.0` since it contains a major fix in it.

**Review by UsamaSadiq — COMMENTED** [review]
This PR should also be ready for the upstream PR.

File: problem_builder/tests/unit/test_answer_mixin.py (hunk 1 above)
Comment (irtazaakram) [reply problem_builder/tests/unit/test_answer_mixin.py:28]: https://github.com/django/django/blob/f474ba4cb52a41a62da6d6fb3492a95eeb8b3999/django/utils/crypto.py#L51

bit length is required argument for `get_random_string` in 4.2
